R/workflow.R

```diff
@@ -15,67 +14,105 @@ Workflow <- R6::R6Class(
-        logDebug(message = workflowFolderCheck)
+        logWorkflow(
+          message = workflowFolderCheck,
+          pathFolder = getwd(),
+          logTypes = c(LogTypes$Debug, LogTypes$Error)
+        )
       }
       self$workflowFolder <- workflowFolder
+      dir.create(self$workflowFolder)
 
-      reportFolder <- file.path(workflowFolder, reportFolderName)
-      reportFolderCheck <- checkExisitingPath(reportFolder, stopIfPathExists = TRUE)
-      if (!is.null(reportFolderCheck)) {
-        logDebug(message = reportFolderCheck)
-      }
-      self$reportFolder <- reportFolder
+      logWorkflow(
+        message = self$reportingEngineInfo$print(),
+        pathFolder = self$workflowFolder
+      )
+
+      self$reportFileName <- file.path(self$workflowFolder, paste0(reportName, ".Rmd"))
 
       resultsFolder <- file.path(workflowFolder, resultsFolderName)
       resultsFolderCheck <- checkExisitingPath(resultsFolder, stopIfPathExists = TRUE)
       if (!is.null(resultsFolderCheck)) {
-        logDebug(message = resultsFolderCheck)
+        logWorkflow(
+          message = resultsFolderCheck,
+          pathFolder = self$workflowFolder,
+          logTypes = c(LogTypes$Debug, LogTypes$Error)
+        )
       }
       self$resultsFolder <- resultsFolder
-
-      figuresFolder <- file.path(workflowFolder, figuresFolderName)
-      figuresFolderCheck <- checkExisitingPath(figuresFolder, stopIfPathExists = TRUE)
-      if (!is.null(figuresFolderCheck)) {
-        logDebug(message = figuresFolderCheck)
-      }
-      self$figuresFolder <- figuresFolder
-
-      # Create workflow output structure
-      createFolder(self$workflowFolder)
-      createFolder(self$reportFolder)
-      createFolder(self$resultsFolder)
-      createFolder(self$figuresFolder)
+      dir.create(self$resultsFolder)
 
       self$simulationStructures <- list()
       # Check of Workflow inputs
-      for (i in 1:length(simulationSets)) {
-        self$simulationStructures[[i]] <- SimulationStructure$new(
-          simulationSet = simulationSets[[i]],
+      for (simulationSetIndex in 1:length(simulationSets)) {
+        self$simulationStructures[[simulationSetIndex]] <- SimulationStructure$new(
+          simulationSet = simulationSets[[simulationSetIndex]],
           workflowResultsFolder = self$resultsFolder
         )
       }
+    },
+
+    #' @description
+    #' Get a vector with all the names of the tasks within the `Workflow`
+    #' @return Vector of `Task` names
+    getAllTasks = function() {
+      # get isTaskVector as a named vector
+      isTaskVector <- unlist(eapply(self, function(x) {
+        isOfType(x, "Task")
+      }))
+
+      taskNames <- names(isTaskVector[as.logical(isTaskVector)])
+
+      return(taskNames)
+    },
+
+    #' @description
+    #' Get a vector with all the names of active tasks within the `Workflow`
+    #' @return Vector of active `Task` names
+    getActiveTasks = function() {

```

Comment:
maybe a private method that can be called by getActiveTasks and getInactiveTasks
The code is exactly the same. 
getTasksWithStatus(bool status) and then you have in your code 

if (self[[taskName]]$active==status) {

Reply:
Great idea for preventing duplication of code
By the way, I feel I am currently duplicating/triplicating quite a few chunks of code for the plot tasks of the mean model workflow:
The steps seem kind of generic :
1) Initialization of the task
- Logging and writing the section title of the report
- Check that the input is available 
- Create an output folder to save the results
2) Run the task
- Logging
- Call the right functions with the right options (this is where they mainly differ) 
3) Termination of the task
- Logging
- Saving the output plots/results of the task
- Writing the report chunks

It may be worth creating a subclass "PlotTask".
So within the Workflow run, I can loop through all tasks with this subclass, to prevent re-using 4 times similar chunks of code.